topology/probes/opencontrail/routing_table.go

```diff
@@ -0,0 +1,339 @@
+// +build linux,opencontrail
+
+/*
+ * Copyright (C) 2018 Orange, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+// When an interface node is created, the VRFID is get from the
+// Contrail Vrouter Agent and associated to this node. This VRF is
+// then dumped (with rt --dump) to populate the Contrail.RoutingTable
+// metadata.
+//
+// The process rt --monitor is spawn to get route update notifications
+// from the Contrail vrouter kernel module. All route updates contain
+// the VRFID. This VRFID is then used to get all interface nodes that
+// have this VRFID. The Contrail routing table of these nodes is then
+// updated according to the route update.
+//
+// LIMITATION: if the Contrail Vrouter Agent is restated, Skydive
+// routing tables are corrupted. Skydive agent then have to be
+// restarted when Contrail Vrouter agent is restarted.
+
+package opencontrail
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/skydive-project/skydive/filters"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+// This represents the data we get from rt --monitor stdout
+type rtMonitorRoute struct {
+	Operation string
+	Family    string
+	VrfId     int `json:"vrf_id"`
+	Prefix    int
+	Address   string
+	NhId      int `json:"nh_id"`
+}
+
+const afInetFamily string = "AF_INET"
+
+const OpenContrailRouteProtocol int64 = 200
+
+// The skydive representation of a Contrail route
+type OpenContrailRoute struct {
```

Comment:
It could be nice to add the following comment
// easyjson:json

modifying the Makefile too. It will bring better performance for serialization thing.

please have a look at

https://github.com/skydive-project/skydive/blob/master/Makefile#L49
https://github.com/skydive-project/skydive/blob/master/topology/probes/netlink/netlink.go#L97

Consider using this every time it is possible